packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -1216,6 +1227,7 @@ export async function runCustomInstallCommand({
     env: spawnOpts?.env || {},
     packageJsonEngines: packageJson?.engines,
     turboSupportsCorepackHome,
+    detectedLockfile: lockfilePath ? path.basename(lockfilePath) : undefined,
```

Comment:
This logic likely needs to be the same in every place where we use it, can we extract this into a function for single point of truth?

Reply:
This way we can also test that we get the correct lockfile in various cases.

Reply:
If there is a reason that each of these instances should vary separately from one another, I'm happy to remove the changes needed review.

Reply:
Added this logic to `scanParentDirs()` so it only happens in one place!